repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -1571,6 +1574,14 @@ private EntityMutationResponse deleteVertices(Collection<AtlasVertex> deletionCa
         }
 
         for (AtlasEntityHeader entity : req.getDeletedEntities()) {
+            String handler;
+            if (categoryEntityGuids.contains(entity.getGuid())) {
```

Comment:
Instead of extracting from vertex & storing Category Guids at 1560,
Just add Type check on entity here